fix(compiler-cli): detect uninvoked functions in defer trigger expressions

SkyZeroZx opened the pull request:
Wrap `@defer` trigger expressions (`when`, `prefetch when`, `hydrate when`) in a conditional context within the TCB to enable TypeScript's TS2774 diagnostic for detecting functions used without invocation.

Previously, signals and functions passed to `when` triggers without parentheses would silently evaluate to truthy, causing unexpected behavior.
Now the compiler reports an error when a function is used as a condition without being called.
 

Example

Before: No warning for this code
```ts
@Component({
  template: `
    @defer (hydrate when flag) {
      <div>Content</div>
    }
  `
})
class MyComponent {
  flag = signal(false);  // Signal not invoked - no warning
}
```
Now: Warning is now shown
```ts
@Component({
  template: `
    @defer (hydrate when flag) {  //  Warning: flag is a function and should be invoked: flag()
      <div>Content</div>
    }                     
  // Error: This condition will always return true since this function is always defined.
  `
})
class MyComponent {
  flag = signal(false);
}
```

JeanMeche commented:
I'm wondering if that usecase should be covered at the TCB level. (ie passing a condition that is always truthy should raise `TS2774`)

SkyZeroZx committed:
fix(compiler-cli): detect uninvoked functions in defer trigger expressions

Wrap `@defer` trigger expressions (`when`, `prefetch when`, `hydrate when`)
in a conditional context within the TCB to enable TypeScript's TS2774
diagnostic for detecting functions used without invocation.

Previously, signals and functions passed to `when` triggers without
parentheses would silently evaluate to truthy, causing unexpected behavior.
Now the compiler reports an error when a function is used as a condition
without being called.
